app/assets/javascripts/admin/addon/services/site-setting-change-tracker.js

```diff
@@ -1,28 +1,30 @@
 import { tracked } from "@glimmer/tracking";
 import Service, { service } from "@ember/service";
 import { TrackedSet } from "@ember-compat/tracked-built-ins";
+import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
 import SiteSetting from "admin/models/site-setting";
+import SiteSettingDefaultCategoriesModal from "../components/modal/site-setting-default-categories";
 
 export default class SiteSettingChangeTracker extends Service {
+  @service dialog;
   @service modal;
 
   @tracked dirtySiteSettings = new TrackedSet();
 
-  add(settingComponent) {
-    this.dirtySiteSettings.add(settingComponent);
+  add(setting) {
+    this.dirtySiteSettings.add(setting);
   }
 
-  remove(settingComponent) {
-    this.dirtySiteSettings.delete(settingComponent);
+  remove(setting) {
+    this.dirtySiteSettings.delete(setting);
   }
```

Comment:
Tracking model instances instead of component instances.

<img width="279" alt="Screenshot 2025-04-02 at 5 58 11 PM" src="https://github.com/user-attachments/assets/d02991e5-6fc0-4fe0-a94c-76151b0ec4a3" />